File: examples/python/http_client.py
from longbridge.openapi import HttpClient, OAuthBuilder

oauth = OAuthBuilder("your-client-id").build(
    lambda url: print(f"Open this URL to authorize: {url}")
)
http_cli = HttpClient.from_oauth(oauth)
resp = http_cli.request(
    "get",
    "/v1/trade/execution/today",
)
print(resp)

File: examples/nodejs/http_client.js
const { HttpClient, OAuthBuilder } = require("longport");

async function main() {
  const oauth = await OAuthBuilder.build("your-client-id", (url) => {
    console.log("Open this URL to authorize: " + url);
  });
  let cli = HttpClient.fromOAuth(oauth);
  let resp = await cli.request("get", "/v1/trade/execution/today");
  console.log(resp);
}

Promise.all([main()]).catch((err) => console.error(err));
